src/olympia/translations/tests/test_models.py

```diff
@@ -734,38 +734,103 @@ def test_forbidden_tags(self):
+        assert str(translation) == 'a text  link'
+
+    def test_uppercase(self):
+        value = (
             'a <a href="http://example.com">link</a> with markup, a text '
-            'http://example.com link, <b>with forbidden tags</b>, '
-            '<script>forbidden tags</script> and <http://bad.markup.com'
+            'http://example.com link, and some raw <b>html://</b>. I <3 HTTP!'
         )
-        x = NoLinksNoMarkupTranslation(localized_string=s)
-        assert x.__html__() == (
-            'a  with markup, a text  link, '
-            '&lt;b&gt;with forbidden tags&lt;/b&gt;, '
-            '&lt;script&gt;forbidden tags&lt;/script&gt; and'
+        translation = NoURLsTranslation(localized_string=value)
+        assert str(translation) == (
+            'a <a href="">link</a> with markup, a text '
+            ' link, and some raw <b>html://</b>. I <3 HTTP!'
+        )
+
+    def test_with_newlines(self):
+        value = (
+            'a <a href="http://example.com">link</a> with markup \n, a text '
+            'http://example.com link, and some raw <b>HTML</b>. I <3 http!'
+        )
+        translation = NoURLsTranslation(localized_string=value)
+        assert str(translation) == (
+            'a <a href="">link</a> with markup \n, a text '
+            ' link, and some raw <b>HTML</b>. I <3 http!'
         )
 
+    def test_bit_of_everything(self):
+        value = (
+            'a <a href="http://example.com">link</a> with markup,\n'
+            'a newline with a more complex link <a href="http://foo.is">lol.com</a>,\n'
+            'another http://example.com link, <b>with forbidden tags</b>,\n'
+            'and <script>forbidden tags</script> as well as <http:/bad.markup.com'
+        )
+        translation = NoURLsTranslation(localized_string=value)
+        assert str(translation) == (
+            'a <a href="">link</a> with markup,\n'
+            'a newline with a more complex link <a href=""></a>,\n'
+            'another  link, <b>with forbidden tags</b>,\n'
+            'and <script>forbidden tags</script> as well as <'
+        )
+
+    def test_clean_on_save(self):
+        translation = NoURLsTranslation.objects.create(id=1001, localized_string='foo')
+        assert (
+            translation.localized_string_clean == translation.localized_string == 'foo'
+        )
+
+        translation = NoURLsTranslation.objects.create(
+            id=1002, localized_string='foo', localized_string_clean='bar'
+        )
+        assert (
+            translation.localized_string_clean == translation.localized_string == 'foo'
+        )
+
+        translation = NoURLsTranslation.objects.create(
+            id=1003,
+            localized_string='foo http://example.com',
+            localized_string_clean='foo',
+        )
+        assert translation.localized_string == 'foo http://example.com'
+        assert translation.localized_string_clean == 'foo'
```

Comment:
Is this testing that `localized_string` has been cleaned, and the URL removed, or is this testing that the value provided for `localized_string_clean` ends up in `translation.localized_string_clean`? I believe it is the former, and if that's the case, perhaps it would be better if the original value passed into `localized_string_clean` is something other than `'foo'`.

Reply:
4313e1e2adb4fb2e69fc176537a2b31026783000